File: android/core-model/src/main/java/de/vibecode/crisis/core/model/CrisisEvent.kt
package de.vibecode.crisis.core.model

import kotlinx.datetime.Instant
import kotlinx.serialization.Serializable

@Serializable
enum class CrisisCategory {
    GEOPOLITICAL,
    FINANCIAL
}

@Serializable
data class CrisisEvent(
    val id: String,
    val title: String,
    val summary: String?,
    val region: String,
    val occurredAt: Instant,
    val publishedAt: Instant?,
    val detailUrl: String?,
    val sourceName: String?,
    val source: DataSource,
    val category: CrisisCategory,
    val severityScore: Double
)

File: CRISIS/CRISIS/Models/CrisisEvent.swift
import Foundation

enum CrisisCategory: String, Codable, CaseIterable {
    case geopolitical
    case financial
}

struct CrisisEvent: Identifiable, Hashable, Codable {
    let id: String
    let title: String
    let summary: String?
    let region: String
    let occurredAt: Date
    let publishedAt: Date?
    let detailURL: URL?
    let sourceName: String?
    let source: DataSource
    let category: CrisisCategory
    let severityScore: Double

    var severityBadge: String {
        switch severityScore {
        case ..<4: return "Info"
        case 4..<6: return "Moderat"
        default: return "Hoch"
        }
    }
}
